### DISCUSSION TITLE
Anyone else have issues with Shiny.AndroidExtensions with 2.5.1?

### DISCUSSION THREAD

**Discussion by ryanjkendrick** [shinyorg/shiny#998] [Q&A] [answered]
Since updating to the very awesome Shiny 2.5.1, I am having issues with the Android counterpart of my Xamarin.Forms application. 

- MainActivity.ShinyOnCreate()
- MainActivity.ShinyOnNewIntent()
- MainActivity.ShinyOnActivityResult()
- MainActivity.ShinyOnRequestPermissionsResult()
- MainApplication.ShinyOnCreate()

All the above cannot be found. The Shiny namespace gets resolved and other classes and interfaces within the Shiny framework can be accessed in this project but those extension methods cannot. This occurs with or without Shiny source generators.

I cannot replicate this issue using the Shiny samples and cannot see what is different. Is anyone else having this problem?

**Comment by aritchie** [comment, marked as answer]
Please make sure to 
a) read the release notes 
b) check the nuget targets

This release bumped the android major target to 12.

**Reply by ryanjkendrick** [reply]
I had read the release notes as well as the brilliant new docs.

I have also checked the nuget targets and they are the same across iOS and Android.

I have managed to reproduce this problem in a blank Xamarin Forms project that I have uploaded here:
[ShinyTest](https://github.com/rhinohq/ShinyTest)

This, perhaps, should have been listed under the 'Issues' section of the Shiny repo.

**Reply by aritchie** [reply]
It's not a bug!  You aren't targeting Android 12 - https://github.com/rhinohq/ShinyTest/blob/eed9e6b49684b9df8034968fdba11138662899aa/ShinyTest.Android/ShinyTest.Android.csproj#L19

**Reply by ryanjkendrick** [reply]
I am sorry, I thought the target SDK was derived from the AndroidManifest.xml in Xamarin.

Thank you very much from your help. 🙂
